examples/submodule_firewall/outputs.tf

```diff
@@ -28,3 +28,48 @@ output "admin_ranges" {
   description = "Firewall attributes for admin ranges."
   value       = module.test-firewall-submodule.admin_ranges
 }
+
+output "network_self_link" {
+  value       = module.test-vpc-module.network_self_link
+  description = "The URI of the VPC being created"
+}
+
+output "svpc_host_project_id" {
+  value       = module.test-vpc-module.svpc_host_project_id
+  description = "Shared VPC host project id."
+}
+
+output "subnets_names" {
```

Comment:
Output names plural_plural feels a bit weird. Any way we could consolidate those outputs (names, ips, regions, private_access, flow_logs, secondary_range into a map indexed by subnet name ?

Something like:
```
{
   subnet_1: {
      ip: 10.0.0.0/19
      region: us-east1
      private_access: true
      flow_logs: true
   },
   subnet_2: {
       ...
   }
}
```

Reply:
I like this idea a lot. For this PR I was mostly just trying not to break the outputs with the change.  Would you prefer me to take a swing at adding a new output to this PR, or do it in a separate one?

Reply:
+1

Reply:
+1 on changing the output format of the main module, I kind of fail to grasp the usefulness of adding subnet outputs to the firewall submodule.

Reply:
@ludoo yeah your right those aren't really needed in that example, I'll remove em

Reply:
@ludoo ahh now I recall, if those outputs aren't defined the lint CI step will fail with these errors. 

`make docker_test_lint`

```
terraform_validate ./test/fixtures/submodule_firewall

Error: Reference to undeclared output value

  on outputs.tf line 33, in output "output_network_self_link":
  33:   value       = module.example.network_self_link

An output value with the name "network_self_link" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 38, in output "output_svpc_host_project_id":
  38:   value       = module.example.svpc_host_project_id

An output value with the name "svpc_host_project_id" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 43, in output "output_subnets_names":
  43:   value       = module.example.subnets_names

An output value with the name "subnets_names" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 48, in output "output_subnets_ips":
  48:   value       = module.example.subnets_ips

An output value with the name "subnets_ips" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 54, in output "output_subnets_regions":
  54:   value       = module.example.subnets_regions

An output value with the name "subnets_regions" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 59, in output "output_subnets_private_access":
  59:   value       = module.example.subnets_private_access

An output value with the name "subnets_private_access" has not been declared
in module.example.


Error: Reference to undeclared output value

  on outputs.tf line 64, in output "output_subnets_flow_logs":
  64:   value       = module.example.subnets_flow_logs

An output value with the name "subnets_flow_logs" has not been declared in
module.example.


Error: Reference to undeclared output value

  on outputs.tf line 69, in output "output_subnets_secondary_ranges":
  69:   value       = module.example.subnets_secondary_ranges

An output value with the name "subnets_secondary_ranges" has not been declared
in module.example.


Error: Reference to undeclared output value

  on outputs.tf line 74, in output "output_routes":
  74:   value       = module.example.routes

An output value with the name "routes" has not been declared in
module.example.

terraform_validate ./test/setup
Success! The configuration is valid.

make: *** [docker_test_lint] Error 1
```

Reply:
@tfhartmann Just keep them like this and change the output format in a separate PR (or open an issue). @ludoo, thoughts ?

Reply:
Yes, let's keep this the same and not change output formats in this PR.

Reply:
Create this issue to track the above proposal about adding a new output map. https://github.com/terraform-google-modules/terraform-google-network/issues/77 . I'm also going to resolve this conversation. Please feel free to re-open if i've closed it in error.